File: C#2/Homeworks/06-Strings-and-Text-Processing/12ParseUrl/ParseUrl.cs
using System;
using System.IO;

/*Write a program that parses an URL address given in the format: [protocol]://[server]/[resource]
and extracts from it the  [protocol] ,  [server]  and  [resource]  elements.

Input
•On the only line you will receive an address

Output
•Print the protocol, server and resource as shown below

Constraints
•Time limit: 0.1s
•Memory limit: 16MB

 http://telerikacademy.com/Courses/Courses/Details/212 
[protocol] = http
[server] = telerikacademy.com
[resource] = /Courses/Courses/Details/212  
 
 */


class ParseUrl
{
   static void Main()
   {
       string url = Console.ReadLine();

       string protocol = "[protocol] = ";
       string server = "[server] = ";
       string resource = "[resource] = ";

       int indexEndProtocol = url.IndexOf("://");
       protocol += url.Substring(0, indexEndProtocol);

       Console.WriteLine(protocol);

       //server
       char slash = Path.AltDirectorySeparatorChar; // '/'
       int indexStartServer = indexEndProtocol + 3; // server part start after 2 slash after the protocol
       int indexEndServer = url.IndexOf(slash,indexStartServer);
       server += url.Substring(indexStartServer, indexEndServer - indexStartServer);

       Console.WriteLine(server);

       //resource
       resource += url.Substring(indexEndServer);
       Console.WriteLine(resource);
   }
}

File: JS/HW/Strings/parse-url.js

// Write a script that parses an URL address given in the format: 
// [protocol]://[server]/[resource] and extracts from it the [protocol], [server] and [resource] elements.

// The input will consist of an array containing one string
// The output should be consisted of three line

function parseUrl(args){

    var url = args[0],
        protocol ="", server = "", resource = "";

    var indexEndProtocol = url.indexOf("://");
    protocol = url.substring(0,indexEndProtocol);

    var indexEndServer = url.indexOf("/",indexEndProtocol+3);
    server = url.substring(indexEndProtocol+3,indexEndServer);

    resource = url.substring(indexEndServer);

    console.log("protocol: " + protocol);
    console.log("server: " + server);
    console.log("resource: " + resource);
}

parseUrl([ 'http://telerikacademy.com/Courses/Courses/Details/239' ]);
//protocol: http
// server: telerikacademy.com
// resource: /Courses/Courses/Details/239